### DISCUSSION TITLE
Resizing windows via the command line

### DISCUSSION THREAD

**Discussion by lamchau** [rxhanson/Rectangle#206] [Ideas]
I love using Rectangle when my machine is docked but have a hard time finding keybindings that work when switching between an external keyboard and the laptop. Currently, I have them mapped to the NumPad since the spatial layout fits perfectly but the differing key codes aren't portable when undocking.

I've been exploring ways to accomplish this with AppleScript and Hammerspoon but thought it'd be better as a first-class feature. So instead, we could provide additional command-line hooks to invoke the underlying commands that are mapped. 

This would unlock a few different novel use cases
- Scripting with a launcher (e.g. Alfred; auto-layout common apps)
- Scripting with a variety of other languages that can interop
- Custom touch bar layouts to display the different layouts

I'm not sure the work involved if we can invoke it directly or need to a separate helper app (`subl` one just for launching Sublime Text from the command-line).

Not quite sure how to get started on this myself (I'm new to Mac development) but would be happy to contribute if I could find a good place to start!

**Comment by rxhanson** [comment]
I did start taking a look at doing this (https://github.com/rxhanson/Rectangle/issues/147), and it got pretty painful pretty fast. 

As is the case for a lot of things, the rey wenderlich tutorial isn't bad: https://www.raywenderlich.com/1033-making-a-mac-app-scriptable-tutorial. I'd suggest starting there first. If you can find a better reference for doing this, definitely let me know.

SizeUp is a pretty good example of this done nicely - I like the way they set up their window actions as an enumeration. In order to achieve that, it would be some pretty painful trial and error programming, though, and that's why I stopped - too time consuming. 

Since Rectangle is set up in a pub-sub way, this technically shouldn't be that hard to do. Honestly, I'd be good with merging a PR for this even if it is clunky and just uses what that ray wenderlich example does rather than the nice enumeration style of doing it that SizeUp did.

From my perspective, this is a challenging first step into mac development but I'm positive it would be a rewarding experience if you got some stuff working.

**Comment by rxhanson** [comment]
Forgot to mention that it makes the most sense to me to have this as part of the application, not as a separate helper app like sublime.
